run.sh

```diff
@@ -10,45 +10,72 @@
 # or in the "license" file accompanying this file. This file is distributed
 # on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 # express or implied. See the License for the specific language governing
-# permissions and limitations under the License.
-#
+# permissions and limitations under the License.  # set +eux
 
-set +eux
+set -eu
+IMGNAME=amzn2-kvm-2.0.20240419-x86_64.qcow2
+#In MB
+MEMORY=8096
 
-if [[ ! -f ~/.ssh/id_ed25519 ]]; then
-  echo "Installing private ssh key for vm"
-  cp ./id_ed25519.txt ~/.ssh/id_ed25519
-  chmod 600 ~/.ssh/id_ed25519
-  cat known_hosts >> ~/.ssh/known_hosts
-fi
+populate_key() {
+    #TODO: get these from Secrets Manager
+    if [[ ! -f ~/.ssh/id_ed25519 ]]; then
+        echo "Installing private ssh key for vm"
+        cp ./id_ed25519.txt ~/.ssh/id_ed25519
+        chmod 600 ~/.ssh/id_ed25519
+        cat known_hosts >> ~/.ssh/known_hosts
+    fi
+}
 
-    #-nographic \
-echo "Launching AL2 image"
-qemu-system-x86_64 -m 4096 \
-    -smp cpus=8 \
-    -display none \
-    -device e1000,netdev=net0 \
-    -netdev user,id=net0,hostfwd=tcp::2222-:22 \
-    -hdb seed.img \
-    -hda amzn2-kvm-2.0.20230628.0-x86_64.xfs.gpt.qcow2 &
-# TODO: Validate disk perf options
-# TODO: compress/clean qcow2 image
+launch () {
+    KVM_ARGS=""
+    if [[ -e /dev/kvm ]]; then
+        KVM_ARGS+="-smp cpus=8 -accel kvm -cpu host"
```

Comment:
Why are we able to hardcode the cpus? Should we be getting this number from the OS?

Reply:
On an i3.metal, with 72 cores, 8 was enough for a reasonable runtime.  KVM isn't available for our regular CI runs, so this won't actually get used in CI (until there is a KVM offering).